runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -238,7 +230,19 @@ public void onTimer(
         String timerId, BoundedWindow window, Instant timestamp, TimeDomain timeDomain) {}
 
     @Override
-    public void finishBundle() {}
+    public void finishBundle() {
+      try {
+        // TODO: it would be nice to emit results as they arrive, can thread wait non-blocking?
+        // RemoteBundle close blocks until all results are received
+        remoteBundle.close();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+      KV<String, OutputT> result;
+      while ((result = outputQueue.poll()) != null) {
+        outputManager.output(outputMap.get(result.getKey()), (WindowedValue) result.getValue());
```

Comment:
This will block forwarding the results until the bundle is finalized. Depending on the bundle size, this could lead to memory pressure and increased latency. Needs to be optimized (probably ok for now).

Reply:
+1 see latest change